types/index.d.ts

```diff
@@ -0,0 +1,195 @@
+declare module "react-sweet-state" {
+  import { ComponentType, ReactNode, ReactElement } from "react";
+
+  type SetState<TState> = (newState: Partial<TState>) => void;
+  type GetState<TState> = () => Readonly<TState>;
+  type StoreUnsubscribe = () => void;
+
+  type RenderPropComponent<TState, TActions> = (
+    state: TState,
+    actions: TActions
+  ) => ReactNode;
+
+  type HookReturnValue<
+    TState,
+    TActions extends Record<string, ActionThunk<TState, TActions>>
+  > = [TState, BoundActions<TState, TActions>];
+
+  type ActionThunk<TState, TActions> = (...args: any[]) => Action<TState>;
+
+  export type Store<
+    TState,
+    TActions extends Record<string, ActionThunk<TState, TActions>>
+  > = {
+    key: string[];
+    initialState: TState;
+    actions: TActions;
+  };
+
+  export type StoreState<TState> = {
+    getState: GetState<TState>;
+    setState: SetState<TState>;
+    key: string[];
+    subscribe: (listener: () => void) => StoreUnsubscribe;
+    mutator: SetState<TState>;
+  };
+
+  export type Action<TState, TContainerProps = void, TActions = any> = (
+    state: {
+      setState: SetState<TState>;
+      getState: GetState<TState>;
+      actions: TActions;
```

Comment:
The `actions` object type is actually `BoundActions<TState, TActions>`. 
I've tried to fix that but TS complains that "Type 'TActions' does not satisfy the constraint Record<...>" which I believe is triggered by the fact that `TActions = any` in the Action type definition, so we are missing the constrain there. But adding it triggers a lot of other errors so not sure which way is the correct way to fix it

Reply:
I'm looking at it :)

Reply:
this should look better.